lib/IDE/CommentConversion.cpp

```diff
@@ -496,6 +496,33 @@ bool ide::getDocumentationCommentAsXML(const Decl *D, raw_ostream &OS,
   return true;
 }
 
+bool ide::getRawDocumentationComment(const Decl *D, raw_ostream &OS) {
+  ClangNode MaybeClangNode = D->getClangNode();
+  if (MaybeClangNode) {
+    const clang::Decl *CD = MaybeClangNode.getAsDecl();
+    if (!CD) {
+      return false;
+    }
+    clang::ASTContext &astCtx = CD->getASTContext();
+    const clang::RawComment *rawComment = astCtx.getRawCommentForAnyRedecl(CD);
```

Comment:
Does the `FullComment` not have the raw comment? Seems like it would be nicer to share implementation between the two raw vs XML functions if we can. This also doesn't do `replaceObjcDeclarationsWithSwiftOnes`, which... maybe matters and maybe doesn't?

Reply:
Changed it to go through the same path that the old implementation did, which has some sophisticated logic in clang’s `ASTContext::getCommentForDecl`.